SignalMessaging/profiles/OWSProfileManager.m

```diff
@@ -765,15 +765,33 @@ - (nullable NSString *)profileNameForRecipientId:(NSString *)recipientId
     return userProfile.profileName;
 }
 
-- (nullable UIImage *)profileAvatarForRecipientId:(NSString *)recipientId
+- (nullable UIImage *)profileAvatarImageForRecipientId:(NSString *)recipientId
 {
     OWSAssert(recipientId.length > 0);
 
     OWSUserProfile *userProfile =
         [OWSUserProfile getOrBuildUserProfileForRecipientId:recipientId dbConnection:self.dbConnection];
 
     if (userProfile.avatarFileName.length > 0) {
-        return [self loadProfileAvatarWithFilename:userProfile.avatarFileName];
+        return [self loadProfileAvatarImageWithFilename:userProfile.avatarFileName];
+    }
+
+    if (userProfile.avatarUrlPath.length > 0) {
```

Comment:
Why this change?

Reply:
If we're going to use a profile avatar, we want to access it as _data_, not as an image to avoid an unneccesary and lossy conversion.